### PR TITLE
ShellCheck plugin simplification and cleanups

### PR DESCRIPTION
**Pull request by ZoogieZork**
## Description

* Pass the `-S` (set minimum severity level) option to ShellCheck so we don't need to filter out findings ourselves.
* Use native Python to get the list of files rather than an `os.subprocess` call to `find`.
* Added unit test for `get_files()`.
* Added types.
* Handle the error case if the call to `shellcheck` fails.

## Motivation and Context

Cleans up several issues with the ShellCheck plugin noted over time.

## How Has This Been Tested?

Tested in nonprod environment.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist

- [x] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/engine/plugins/shell_check/main.py [hunk 1]
```diff
@@ -3,45 +3,38 @@
...
+import pathlib
 import subprocess
+from typing import Any, Optional
 
 from engine.plugins.lib import utils
 
 logger = utils.setup_logging("shell_check")
 
+# Filename extensions to include in scanning.
+INCLUDE_EXTS = {".sh", ".bash", ".ksh", ".bashrc", ".bash_login", ".bash_logout", ".bash_profile", ".bash"}
```
Comment (g-marconet) [inline backend/engine/plugins/shell_check/main.py:15]: Nit: `.bash` is repeated

**Review by g-marconet — COMMENTED** [review]
Looks good, just one small non-functional thing

File: backend/engine/plugins/shell_check/main.py (hunk 1 above)
Comment (ZoogieZork) [reply backend/engine/plugins/shell_check/main.py]: Good catch!  Fixed.

**Review by g-marconet — APPROVED** [review]
LGTM

**Pull request merged**